base/shared/src/main/scala/scalaz/typeclass/SemigroupClass.scala

```diff
@@ -1,6 +1,6 @@
 package scalaz
 package typeclass
 
-trait SemigroupClass[A] extends Semigroup[A]{
+trait SemigroupClass[A] extends Semigroup[A] with MagmaClass[A]{
```

Comment:
These subclassings need to be reflected in `BaseHierarchy`.

Reply:
even if we consider removing Magma, I added it there already
https://github.com/scalaz/scalaz/pull/1551/commits/316260023ea769db3598e9b9883f10e850b69f36